Support primary_key in mirror tag

kempfsn opened the pull request:
This adds support for a public key of the primary mirror. The primary key is used in the following places:
- to verify the Release of the primary mirror for the initvm
- to add the primary key to the keyring of the initvm
- to call debootstrap with the primary key when building the rootfs
- to call debootstrap with the primary key when creating a pbuilder

Resolves #104 and #365.

d4nuu8 commented:
We are having exactly the same problem. I'm going to test this patch with our code base and give you feedback.

@bgermann Could you take a look at this, too?

bgermann commented:
It is on my todo list.

d4nuu8 commented:
Doesn't work for me unfortunately. But I'm not sure if it is our special corporate environment or the patch. APT is failing to install packages because of verification errors (e.g. busybox). 

I'm missing the time to dig deeper here right now. Sorry.

kempfsn commented:
> Sorry for the long delay. I used this PR, rebased it and cleaned up a few minor parts.
> 
> https://lists.linutronix.de/pipermail/elbe-devel/2024-November/007571.html
> 
> Could you give it a test? @kempfsn I kept your authorship. Could you confirm if you are fine with it?

@t-8ch I do not have this use case anymore and cannot test unfortunately. But would be fine with it.

t-8ch commented:
This has been implemented in d885072a8c253e92223d440a6d5f23def987e5c7..e28d8ad431fb267f20e96ab5dd8172528e6ff69b
